pkg/minikube/notify/notify_test.go

```diff
@@ -0,0 +1,116 @@
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package notify
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"path/filepath"
+	"testing"
+	"time"
+
+	"github.com/blang/semver"
+	"github.com/spf13/viper"
+	"k8s.io/minikube/pkg/minikube/config"
+	"k8s.io/minikube/pkg/minikube/tests"
+	"k8s.io/minikube/pkg/version"
+)
+
+func TestShouldCheckGithubNoWantUpdate(t *testing.T) {
+	viper.Set(config.WantUpdateNotification, false)
+	tempDir := tests.MakeTempDir()
+	tempLastUpdateCheckFilePath := filepath.Join(tempDir, "last_update_check")
+	if shouldCheckGithubVersion(tempLastUpdateCheckFilePath) == true {
+		t.Fatalf("Error: shouldCheckGithubVersion returned true even though config had WantUpdateNotification: false")
+	}
+}
+
+func TestShouldCheckGithubLongerThanWaitPeriod(t *testing.T) {
```

Comment:
Looks like you're missing tests for the file not existing, and the file being malformed. The tests look good though, nice job.